src/test/java/dk/kb/image/FacadeTest.java

```diff
@@ -0,0 +1,101 @@
+package dk.kb.image;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.imageio.ImageIO;
+import dk.kb.util.Resolver;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FacadeTest {
+    private Logger log = LoggerFactory.getLogger(this.toString());
+
+
+    // method to validate that image is greyscale
+    public static boolean isGreyscale(BufferedImage image){
+        // get width and height of image 
+        int width = image.getWidth();
+        int height = image.getHeight();
+        // integers for pixel and colors
+        int pixel,red, green, blue;
+        // Nested loop to check pixels
+        // Works vertically through pixels from bottom left 
+        for (int i = 0; i < width; i++){
+            for (int j = 0; j < height; j++) {
+                // Get RGB value of single pixel
+                pixel = image.getRGB(i, j);
+                // Bitwise rightshifts in colors
+                red = (pixel >> 16) & 0xff;
+                green = (pixel >> 8) & 0xff;
+                blue = (pixel) & 0xff;
+                // Check that colors are equal to eachother
+                // If not equal, the image contains color
+                if (red != green || green != blue ) return false;
+            }
+        }
+        return true;
+    }
+
+    public static boolean canCountColors(BufferedImage image){
```

Comment:
This looks like a near `1:1` re-implementation of the Facade method. So it does not really test anything. As the input image does not change, maybe use another tool (ImageMagick/Gimp) to count the colors and simply put that number into the `testColorCount`-method directly?

Reply:
This is indeed very close to a re-implementation. I believe I wrote this test without fully understanding how to use unit testing. I also believe that I made this test to do some kind of sanity check.
In retrospect this test might not be needed at all?

Reply:
It is needed if we care about the result being correct :-)